app/lib/actions/katello/host/upload_package_profile.rb

```diff
@@ -29,11 +29,7 @@ def rescue_strategy
           Dynflow::Action::Rescue::Skip
         end
 
-        def run
-          profile = JSON.parse(input[:profile_string])
-          #free the huge string from the memory
-          input[:profile_string] = 'TRIMMED'.freeze
-
+        def self.upload(input, profile)
```

Comment:
This should not take 'input' as a param, lets call out the params, otherwise there's no way to know what to pass in.